xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp

```diff
@@ -3021,12 +3020,19 @@ IAESound *CActiveAE::MakeSound(const std::string& file)
 
     // decode until eof
     av_init_packet(&avpkt);
-    int len;
+    int ret = 0;
     while (av_read_frame(fmt_ctx, &avpkt) >= 0)
     {
       int got_frame = 0;
-      len = avcodec_decode_audio4(dec_ctx, decoded_frame, &got_frame, &avpkt);
-      if (len < 0)
+      ret = avcodec_send_packet(dec_ctx, &avpkt);
```

Comment:
I don't like this. a return value of a function should be checked for error right after the call, not 10 lines after.